ParcialProgramacion4/src/Empresa.java

```diff
@@ -58,4 +60,371 @@ private Habilidad buscarHabilidad(String nombre) {
+
+            System.out.println("Empleado agregado a lista de la empresa!!!");
+
+        } else {
+            System.out.println("ERROR: ya existe un empleado con ese numero de legajo");
+        }
+    }
+
+    private ArrayList<Cargo> pedirListaCargos(Fecha fechaIngresoEmpresa) {
+        Scanner scanner = new Scanner(System.in);
+        // INGRESAR TODOS LOS CARGOS QUE UN EMPLEADO OCUPO HASTA AHORA
+
+        System.out.println("\nA continuación ingresara el historial de cargos ocupados por el empleado: ");
+        System.out.println("Primero se pedira los cargos ANTIGUOS que ocupo el mismo (Se ingresa SI en caso de que los tenga)");
+        System.out.println("Luego cuando usted lo decida, ingresara el cargo ACTUAL (Ingresando NO a la pregunta 'tiene puestos antiguos?')\n");
+
+        //creo un arraylist local para el historial de cargos para pasarle al constructor de empleado (es lo que retorno)
+        ArrayList<Cargo> historialDeCargos;
+        historialDeCargos = new ArrayList<Cargo>();
+
+        //primero ingreso los cargos antiguos
+        System.out.println("\nTiene puestos ANTIGUOS? (SI/NO)");
+        String tienePuestoAntiguo = scanner.nextLine();
+        
+        while (!tienePuestoAntiguo.equalsIgnoreCase("SI") && !tienePuestoAntiguo.equalsIgnoreCase("NO")) {
+            System.out.println("Ingrese una opcion valida, SI o NO");
+            System.out.println("Tiene puestos antiguos? (SI/NO)");
+            tienePuestoAntiguo = scanner.nextLine();
+        }
+        
+        
+        
+        //carga del cargo mas antiguo en la que la fecha de ingreso al cargo es la misma que la de ingreso a la empresa
+        //primero es un 'if' y no un while porque la fecha de inicio del cargo mas antiguo se saca de la fecha de ingreso a la empresa
+        //pero en los proximos cargos antiguos se saca de la fecha de fin del anterior cargo
+        if (tienePuestoAntiguo.equalsIgnoreCase("SI")) {
+        	//se repite el codigo porque asi puedo sacar la fecha en la que ingreso a la empresa
+        	//para utilizarla como la fecha de inicio del primer cargo
+        	
+            System.out.println("\nRECORDAR: Los puestos antiguos se deben ingresar en orden empezando desde el mas antiguo\n");
+
+        	System.out.println("Nombre puesto: ");
+            String nombrePuesto = scanner.nextLine();
+
+            Puesto puesto = this.buscarPuesto(nombrePuesto);
+
+            //SI NO EXISTE, SE LE PREGUNTA SI LO QUIERE AGREGAR:
+            if (puesto == null) {
+                puesto = this.agregarPuesto(nombrePuesto); //Adentro le doy la posibilidad de crearlo o no
+            }
+
+            //tal vez el usuario puso que no quiere agregar un puesto, por eso verifico lo mismo
+            if(puesto != null) {
+                //fecha de ingreso al cargo es la fecha en la que ingreso a la empresa, se ingresa como parametro
+
+                System.out.println("\nFecha final del puesto: ");
```

Comment:
Este no estoy tan seguro, porque no se como es el input de fecha
```suggestion
                System.out.print("\nFecha final del puesto: ");
```